agent/src/agent/api/routes/pipeline.py

```diff
@@ -132,3 +133,22 @@ def reset(pipeline_id):
     except StreamSetsApiClientException as e:
         return jsonify(str(e)), 500
     return jsonify('')
+
+
+@pipelines.route('/pipeline-failed', methods=['POST'])
+def pipeline_failed():
+    data = request.get_json()
+    destination = pipeline.repository.get(data['pipeline_title']).destination
+    metric = [{
+        "properties": {
+            "what": "pipelineError",
```

Comment:
- no camelcase please, use underscore
- you definitely don't want time in dimensions
- this metric is a counter, so add `target_type`
- add metadata tags with `pipeline.meta_tags()` instead of setting pipelineTitle
- a more descriptive metric name would be I think `pipeline_error_status_count`